apps/arweave/src/ar_http_iface_middleware.erl

```diff
@@ -2332,62 +2344,22 @@ handle_block_announcement(#block_announcement{ indep_hash = H, previous_block =
 			case ar_node:get_block_shadow_from_cache(PrevH) of
 				not_found ->
 					{412, #{}, <<>>, Req};
-				#block{ height = Height } ->
+				#block{} ->
 					Indices = collect_missing_tx_indices(Prefixes),
-					IsSolutionHashKnown =
```

Comment:
Why are we removing all this code?

Reply:
This one might be questionable, but I felt like this feature is more trouble maintaining than it is actually useful. The feature is about the validators optionally informing the block sender they have the chunks so that some bandwidth is saved while the validator reads and packs the chunk(s) themselves.

Thinking about it now, I believe the feature might be in fact hurtful in the long run especially given the composite packing specifics because it may make miners rely excessively on validators extracting the chunks and if such validators disappear over time the network might not notice it and might not react sufficiently quickly. In other words, I would rather make the network spend some bandwidth but require all miners to set up the chunk building correctly from the start.